dev/tests/api-functional/testsuite/Magento/GraphQl/Swatches/ProductViewTest.php

```diff
@@ -0,0 +1,151 @@
+use Magento\TestFramework\ObjectManager;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+/**
+ * Class ProductViewTest
+ * @package Magento\GraphQl\Swatches
+ */
+class ProductViewTest extends GraphQlAbstract
+{
+    public const IMAGE = 'IMAGE';
+    public const COLOR = 'COLOR';
+    /**
+     * @var array
+     */
+    private $configurableOptions = [];
+    /**
+     * @var SwatchesHelper
+     */
+    private $swatchesHelper;
+
+    /**
+     * @inheritdoc
+     */
+    public function setUp()
+    {
+        $this->swatchesHelper = ObjectManager::getInstance()->get(SwatchesHelper::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Swatches/_files/configurable_products_with_swatches.php
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     */
+    public function testQuerySwatchDataFields()
+    {
+        $productSku = 'configurable';
+
+        $query
+            = <<<QUERY
+{
+  products(filter: {sku: {eq: "configurable"}}) {
+    items {
+      name
+        ... on ConfigurableProduct{
+        configurable_options{
+          id
+          label
+          values {
+            value_index
+            label
+            default_label
+            store_label
+            use_default_value
+          }
+        }
+      }
+    }
+  }
+}
+QUERY;
+        $response = $this->graphQlQuery($query);
+
+        /** @var ProductRepositoryInterface $productRepository */
+        $productRepository = ObjectManager::getInstance()->get(ProductRepositoryInterface::class);
+        $product = $productRepository->get($productSku, false, null, true);
+        /** @var MetadataPool $metadataPool */
+        $metadataPool = ObjectManager::getInstance()->get(MetadataPool::class);
+        $product->setId(
+            $product->getData($metadataPool->getMetadata(ProductInterface::class)->getLinkField())
+        );
+        $this->assertArrayHasKey('products', $response);
+        $this->assertArrayHasKey('items', $response['products']);
+        $configurableOptions = $this->getConfigurableOptions();
+
+        $this->assertBaseFields($response['products'], $configurableOptions);
+    }
+
+    /**
+     * @param $actualResponse
+     * @param $configurableOptions
+     */
+    private function assertBaseFields($actualResponse, $configurableOptions): void
+    {
+        $assertionMap =
+            [
+                'items' => [
+                    [
+                        'name' => 'Configurable Product',
+                        'configurable_options' => [
+                            [
+                                'id' => (int)$configurableOptions['id'],
+                                'label' => $configurableOptions['label'],
+                                'values' => $configurableOptions['options']
+                            ],
+                        ],
+                    ],
+                ],
+            ];
+
+        $this->assertResponseFields($actualResponse, $assertionMap);
```

Comment:
Looks like we can use a few simple `assertEquals` instead of complex `assertResponseFields`